src/erl_optics_foil_server.erl

```diff
@@ -0,0 +1,103 @@
+-module(erl_optics_foil_server).
+-include("erl_optics.hrl").
+
+-behaviour(gen_server).
+
+-export([
+    init/1,
+    handle_cast/2,
+    handle_call/3,
+    handle_info/2,
+    terminate/2,
+    code_change/3
+]).
+
+-export([
+    start_link/0,
+    add_lens/2,
+    stop/0
+]).
+
+-define(SERVER, ?MODULE).
+
+-record(state, {
+    new_lenses  :: list(),
+    timer_ref :: timer:tref() | undefined
+}).
+
+%%%=========
+%%% API
+%%%=========
+
+%Modes: carbon | blank
+
+start_link() ->
+    gen_server:start_link({local, ?SERVER}, ?MODULE, [], []).
+
+stop() ->
+    gen_server:cast(?SERVER, stop).
+
+%%%==========
+%%% Callbacks
+%%%==========
+
+%% TODO: Change the timer to start after a new lens has been
+%%       declared and stop after a while to avoid a permanent
+%%       server doing useless things.
+
+init(_) ->
+    Interval = ?ENV(?ENV_FOIL_RELOAD_INTERVAL, ?DEFAULT_FOIL_RELOAD_INTERVAL),
+    ets:new(erl_optics_ets, [named_table, public, {write_concurrency, false}, {read_concurrency, true}]),
+    {ok, Tref} = timer:send_interval(Interval, reload_foil),
+    {ok, #state{new_lenses = [], timer_ref = Tref}, 0}.
+
+handle_cast({add_lens, Lens}, State) ->
+    {noreply, #state{new_lenses =  [Lens | State#state.new_lenses]}};
+handle_cast(stop, State) ->
+    {stop, normal, State};
+handle_cast(_, State) ->
+    {noreply, State}.
+
+handle_call(reload_foil, _From, State) ->
+    Lenses = State#state.new_lenses,
+    case State#state.new_lenses of
```

Comment:
The handle_call and handle_info versions should probably call the same function to check+insert new_lenses.

Reply:
Done.